xarray/core/dataarray.py

```diff
@@ -2663,8 +2663,15 @@ def func(self, other):
             # don't support automatic alignment with in-place arithmetic.
             other_coords = getattr(other, "coords", None)
             other_variable = getattr(other, "variable", other)
-            with self.coords._merge_inplace(other_coords):
-                f(self.variable, other_variable)
+            try:
+                with self.coords._merge_inplace(other_coords):
+                    f(self.variable, other_variable)
+            except MergeError:
+                raise ValueError(
+                    "Automatic alignment is not supported for in-place operations.\n"
+                    "Consider aligning the indices manually or using a not-in-place operation.\n"
+                    "See https://github.com/pydata/xarray/issues/3910 for more explanations."
+                )
```

Comment:
I think adding `from` here will set `__cause__` properly, so Python will print the nice message "The above exception was the direct cause of the following exception:"
```suggestion
            except MergeError as exc:
                raise MergeError(
                    "Automatic alignment is not supported for in-place operations.\n"
                    "Consider aligning the indices manually or using a not-in-place operation.\n"
                    "See https://github.com/pydata/xarray/issues/3910 for more explanations."
                ) from exc
```